src/json-schema.js

```diff
@@ -26,58 +31,101 @@ function convertSubSchema(schema) {
   }
 
   if (schema.allOf) {
-    actualSchema.allOf = schema.allOf.map(convertSubSchema);
+    actualSchema.allOf = schema.allOf.map(s => convertSubSchema(s, references));
   }
 
   if (schema.anyOf) {
-    actualSchema.anyOf = schema.anyOf.map(convertSubSchema);
+    actualSchema.anyOf = schema.anyOf.map(s => convertSubSchema(s, references));
   }
 
   if (schema.oneOf) {
-    actualSchema.oneOf = schema.oneOf.map(convertSubSchema);
+    actualSchema.oneOf = schema.oneOf.map(s => convertSubSchema(s, references));
   }
 
   if (schema.not) {
-    actualSchema.not = convertSubSchema(schema.not);
+    actualSchema.not = convertSubSchema(schema.not, references);
   }
 
   // Array
 
   if (schema.items) {
     if (Array.isArray(schema.items)) {
-      actualSchema.items = schema.items.map(convertSubSchema);
+      actualSchema.items = schema.items.map(s => convertSubSchema(s, references));
     } else {
-      actualSchema.items = convertSubSchema(schema.items);
+      actualSchema.items = convertSubSchema(schema.items, references);
     }
   }
 
   if (schema.additionalItems && typeof schema.additionalItems === 'object') {
-    actualSchema.additionalItems = convertSubSchema(schema.additionalItems);
+    actualSchema.additionalItems = convertSubSchema(schema.additionalItems, references);
   }
 
   // Object
 
   if (schema.properties) {
     Object.keys(schema.properties).forEach((key) => {
-      actualSchema.properties[key] = convertSubSchema(schema.properties[key]);
+      actualSchema.properties[key] = convertSubSchema(schema.properties[key], references);
     });
   }
 
   if (schema.patternProperties) {
     Object.keys(schema.patternProperties).forEach((key) => {
-      actualSchema.patternProperties[key] = convertSubSchema(schema.patternProperties[key]);
+      actualSchema.patternProperties[key] =
+        convertSubSchema(schema.patternProperties[key], references);
     });
   }
 
   if (schema.additionalProperties && typeof schema.additionalProperties === 'object') {
-    actualSchema.additionalProperties = convertSubSchema(schema.additionalProperties);
+    actualSchema.additionalProperties = convertSubSchema(schema.additionalProperties, references);
   }
 
   return actualSchema;
 }
 
+function lookupReference(reference, root) {
+  const parts = reference.split('/');
+
+  if (parts[0] !== '#') {
+    throw new Error('Schema reference must start with document root (#)');
```

Comment:
I've tested all these error paths at the unit-level. However they shouldn't be possible in real-world because the dereferencer will catch this since only valid circular referenced will make it to lookupReference at the moment.